### PR TITLE
initial attempt at unification

### PR DESCRIPTION
**Pull request by wangfowen**
Code quality not great :/ will think harder about how to structure things nicer, and also write some tests
Currently a bunch of prep items have no name... not sure what's up with that.

### CODE REVIEW

File: client/containers/Recipe.js [hunk 1]
```diff
@@ -40,18 +49,37 @@ class Recipe extends Component {
...
-          <div className="ingredients" style={ingStyle}>
-            <h1 style={h1Style}>Ingredients:</h1>
-            {inventory && inventory.map(i => <Ingredient {...i} />)}
-          </div>
-          <div className="instructions">
-            <h1 style={h1Style}>Method of Prep:</h1>
-            {instructions && instructions.map(i => <Instruction {...i} />)}
-          </div>
+        <div>
+          <MediaQuery print>
```
Comment (sherbondy) [inline client/containers/Recipe.js]: Might just want to use a good-ole stylesheet for this. at least for the `@media print` variant.

File: client/reducers/index.js [hunk 2]
```diff
@@ -26,30 +30,22 @@ const recipes = (state = defaultState, action) => {
...
-        ...state,
-        /*
-   * TODO: is there a nice way to map the key-value pairs for an object or something like that?
-   * basically a clean way to pass to recipe function here,
-   * while also making LOAD_INVENTORY and the mapStateToProps in containers/Recipe.js cleaner
-   */
-        recipes: state.recipes.map(r => recipe(r, action))
-      };
+      Object.keys(state.recipes).forEach(key => {
+        state.recipes[key] = recipe(state.recipes[key], action);
```
Comment (sherbondy) [inline client/reducers/index.js]: Isn't this an instance of directly mutating the state, which is a no-no in redux land and may be the cause of render not triggering properly?

http://redux.js.org/docs/Troubleshooting.html#troubleshooting

File: client/reducers/index.js [hunk 3]
```diff
@@ -26,30 +30,22 @@ const recipes = (state = defaultState, action) => {
...
-        return {
-          ...state,
-          recipes: [
-            ...state.recipes,
-            {
-              id: action.id
-            }
-          ]
+      if (!state.recipes[keyId(action.id, action.isMenuRecipe)]) {
+        state.recipes[keyId(action.id, action.isMenuRecipe)] = {
```
Comment (sherbondy) [inline client/reducers/index.js]: Another instance of direct state mutation!

File: client/reducers/index.js (hunk 2 above)
Comment (wangfowen) [reply client/reducers/index.js]: .... that was exactly it. good catch!

**Comment by sherbondy** [comment]
Nice! This is so close to a usable prototype.
Gonna share preview with Clover corporate team today, awesome job!

**Pull request merged**